### PR TITLE
feat(orgunit-select): only show user-assigned orgunits

### PR DESCRIPTION
**Pull request by Birkbjo**
See https://jira.dhis2.org/browse/DHIS2-6771

Adds support for only selecting user-assigned orgunits, instead of global root-orgunit.
This means that we needed to support multiple roots, as user-assigned OrgUnits can be siblings, not just children of the root. This was done by simply rendering multiple OrgUnit-trees, as I reused the old orgunit tree. However, we may want to update this to the new organisationUnitTree-widget?

There are mulitple "user-assigned" org-units, namely: "Data capture and maintenance organisation units", "Data output and analytic organisation units" and "Search Organisation Units". I am not 100% sure which one to use as the ticket does not specify this, but I reckon it is the second, which is called "dataViewOrganisationUnits" in the API.

Update from Scott: It's actually "data capture and maintenance organisation units".

### CODE REVIEW

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (HendrikThePendric) [inline src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:107]: I think you could simplify this part. Currently you are mapping over the roots and displaying individual `OrgUnitTree` components. 

However, I believe there is also the `OrgUnitTreeMultipleRoots` which does this for you already. That'd be slightly easier.

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (HendrikThePendric) [inline src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:62]: I'm not 100% sure this is always going to behave correctly. If the goal is to present users with the `ALL` authority with all the org-units, which I think is correct, then I would consider changing this a little bit... Maybe something like this:
```javascript
const fields = 'id,displayName,path,children::isNotEmpty,memberCount';

return d2.currentUser.authorities.has('ALL')
    ? d2.models.organisationUnits.list({ paging: false, level: 1, fields })
    : d2.currentUser.getDataViewOrganisationUnits({ fields, paging: false, })

```

**Review by HendrikThePendric — COMMENTED** [review]
Code is looking good. I've left a few comments, but didn't formally request changes because my holiday is starting tomorrow and I don't want to block this PR.

> but I reckon it is the second, which is called "dataViewOrganisationUnits" in the API.

I agree with that

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (Birkbjo) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:62]: Why would it not behave correctly? It is only for admin users that does not have assigned orgunits. Eg. `system`-user. If the user has assigned org-units and has the all-authority we should just use the assigned org-units, at least that's how it has worked in the maintenance app for ever.

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (Birkbjo) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:107]: Yeah wasn't aware of that actually, but it does basically the exact same thing when looking at the code (except with some validation for props). I will try to use that.

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (HendrikThePendric) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:62]: > If the user has assigned org-units and has the all-authority we should still use the assigned org-units, at least that's how it has worked in the maintenance app forever.

OK, if this is the case, the current code makes perfect sense.

**Review by HendrikThePendric — APPROVED** [review]
Approving now because the only serious concern I had was addressed in https://github.com/dhis2/data-quality-app/pull/631#discussion_r545167153

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (Birkbjo) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:62]: I will update the comment with a note for that.

File: src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js
Comment (Birkbjo) [reply src/components/available-organisation-units-tree/AvailableOrganisationUnitsTree.js:107]: Actually I will keep it as is, as it doesn't seem possible to initially expand all roots with that component.

**Pull request merged**
